pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -337,6 +339,39 @@ func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneLi
 	return nil
 }
 
+// Dicsover E/P cores
+func discoverCpuCores() v1alpha2.AttributeList {
+	attrList := v1alpha2.AttributeList{}
+
+	cpu_atom, err := os.ReadFile(hostpath.SysfsDir.Path("sys/devices/cpu_atom", "cpus"))
+	if err != nil {
+		klog.ErrorS(err, "error reading cpu_atom/cpus file")
+		return v1alpha2.AttributeList{}
+	}
+
+	attrAtom := v1alpha2.AttributeInfo{
+		Name:  "cpu_atom",
+		Value: strings.TrimSpace(string(cpu_atom)),
+	}
+
+	attrList = append(attrList, attrAtom)
+
+	cpu_core, err := os.ReadFile(hostpath.SysfsDir.Path("sys/devices/cpu_core", "cpus"))
+	if err != nil {
+		klog.ErrorS(err, "error reading cpu_core/cpus file")
+		return v1alpha2.AttributeList{}
+	}
+
+	attrCore := v1alpha2.AttributeInfo{
+		Name:  "cpu_core",
+		Value: strings.TrimSpace(string(cpu_core)),
+	}
```

Comment:
nit/suggestion: Looks like identical functionality to the cpu_atom stuff above. Maybe split into a function or do a for loop. WDYT?